### PR TITLE
Flip Valid Trade OT Compare Operator

### PR DESCRIPTION
**Pull request by ReignOfComputer**
Related:
https://projectpokemon.org/home/forums/topic/42528-bw2-in-game-trades-yancycurtis-20171117/

### CODE REVIEW

**Comment by ReignOfComputer** [comment]
At least, I think this is correct >.<

File: PKHeX.Core/Legality/Checks.cs [hunk 1]
```diff
@@ -419,7 +419,7 @@ private void VerifyTradeOTOnly(string[] validOT)
             if (pkm.IsNicknamed)
                 AddLine(Severity.Invalid, V9, CheckIdentifier.Nickname);
             int lang = pkm.Language;
-            if (validOT.Length >= lang)
```
Comment (evandixon) [inline PKHeX.Core/Legality/Checks.cs]: This looks like a guard for `validOT[lang]`. Wouldn't this cause issues if lang is actually greater than validOT.Length?

Comment (kwsch) [reply PKHeX.Core/Legality/Checks.cs]: it's more obvious if you flip the two arguments around
it's currently flagging if lang is less than, should be flagging if lang >= validOT.Length

**Comment by kwsch** [comment]
test case pkm still fails as the OT is incorrect; need to verify the order of trainer langs for this

**Pull request merged**